Apto/Catalog/Domain/Core/Model/Product/Rule/RuleCriterion.php

```diff
@@ -201,18 +253,44 @@ public function copy(AptoUuid $id, Collection &$entityMapping): ?RuleCriterion
 
         // set section id
         $orgSectionId = $this->getSectionId();
-        /** @var Section|null $section */
-        $section = null === $orgSectionId ? null : $entityMapping->get($orgSectionId->getId());
-        $sectionId = null === $section ? null : $section->getId();
+        $sectionId = null;
```

Comment:
i think we should better fill the mapping array correct from calling place to remove complexity in the copy function itself